{{ cookiecutter.project_slug }}/Makefile

```diff
@@ -1,15 +1,23 @@
-.PHONY: all extract validate transform build check publish clean
+.PHONY: requirements all extract validate transform build check publish clean
+
 
 EXT = txt
-RESOURCE_NAMES := $(shell python main.py resources)
+
+RESOURCE_NAMES := $(shell python main.py resources) && exit 1
 OUTPUT_FILES := $(addsuffix .csv,$(addprefix data/,$(RESOURCE_NAMES)))
 
-all: extract validate transform build check
+all: requirements extract validate transform build check
+
+%: requirements
```

Comment:
Nunca vi esse tipo de target. O que ele está fazendo? Tem um link de onde tirou isso preu ler?

Imagino que você não colocou simplesmente uma chamada

```
CHECK_DEPENDENCIES := $(shell python scripts/requirements.py)
```

porque isso não paralisa a execução do make[^1], correto?


[^1]: Dei uma [pesquisada](https://stackoverflow.com/questions/225542/how-to-make-a-failing-shell-command-interrupt-make) e também não achei nada elegante pra conseguir o efeito de que uma falha na função `shell` paralisa o make.

Reply:
> porque isso não paralisa a execução do make

Foi o melhor jeito que encontrei do make parar completamente em caso de falha e de sempre executar o script requirements.py não importa qual target for executado.

> Tem um link de onde tirou isso preu ler?

Se nem você conhecia é por que eu realmente tive de cavar muito pra achar hehehe.

Fiquei muito tempo pesquisando e usando GPT e fui juntando os efeitos de cada opção até chegar nisso. Tenho de ver meu histórico aqui pra ver se acho onde tenha algo sobre isso.

O `%` é um curinga para qualquer nome de target e ele tem como dependência o target requirements ( `%: requirements`) e o `:` logo em seguida é um comando que serve para que o target em si não faça nada. Assim pra qualquer outro target ele coloca como dependencia  o target `requirements`. A outra opção era fazer isso manualmente e colocar `requirements` na frente de todos os targets do make. Achei essa segunda opção não-chiconless (não elegante).

A forma atual pelo menos é bem minimalista.

> ```
> CHECK_DEPENDENCIES := $(shell python scripts/requirements.py)
> ```

Se colocar somente isso no make não sai nada no output, pelo menos aqui não:

```
make extract
python main.py extract fact && python main.py extract dim && true
2023-10-26T13:45:01-0300 INFO  [scripts.extract] Extract not implemented for resource fact...
2023-10-26T13:45:02-0300 INFO  [scripts.extract] Extract not implemented for resource dim...

```

Reply:
> Se colocar somente isso no make não sai nada no output, pelo menos aqui não:

É porque atualmente você está executando ele via `typer`, pra executar direto do jeito que eu sugeri precisaria de inserir:

```python
if __name__ == '__main__':
    check_requirements()
```

no `scripts/requirements.py`.

Reply:
> É porque atualmente você está executando ele via `typer`, pra executar direto do jeito que eu sugeri precisaria de inserir:

Fiz essas alterações e agora aparece o output. Mas o make ainda não é interrompido usando `CHECK_DEPENDENCIES := $(shell python scripts/requirements.py)`. Você sabe uma forma de fazer essa interrupção com essa abordagem?

O script standalone utilizando o target curinga `%:` continua a interromper o make como antes.